### PR TITLE
Add connectTimeout and readTimeout to ServiceOptions

### PR DESCRIPTION
**Pull request by mziccard**
Follows from #232.

### CODE REVIEW

File: gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java [hunk 1]
```diff
@@ -150,6 +155,16 @@ public B serviceRpcFactory(ServiceRpcFactory<ServiceRpcT, OptionsT> serviceRpcFa
       this.serviceRpcFactory = serviceRpcFactory;
       return self();
     }
+
+    public B connectTimeout(int connectTimeout) {
```
Comment (aozarov) [inline gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java]: lets document the units (milliseconds) and suggest "0" for infinite and negative for default settings.

File: gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java [hunk 2]
```diff
@@ -160,6 +175,8 @@ protected ServiceOptions(Builder<ServiceRpcT, OptionsT, ?> builder) {
     authCredentials = firstNonNull(builder.authCredentials, defaultAuthCredentials());
     retryParams = builder.retryParams;
     serviceRpcFactory = builder.serviceRpcFactory;
+    connectTimeout = builder.connectTimeout;
```
Comment (aozarov) [inline gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java]: getters?

File: gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java (hunk 1 above)
Comment (mziccard) [inline gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java]: I was planning to add some docs for all the methods in the builder, we can make it part of this PR if you don't mind.

Comment (aozarov) [inline gcloud-java-core/src/main/java/com/google/gcloud/ServiceOptions.java]: Works for me.

**Comment by aozarov** [comment]
Looks good. @mziccard I assume you are done with it, right? I don't get notification when you are adding a new commit to a PR (is there a way to do that?), so I was expecting to a PTAL kind of message...

**Comment by mziccard** [comment]
@aozarov Sorry, yes I was done with it :)

**Pull request merged**